src/utils/cpg.ts

```diff
@@ -0,0 +1,263 @@
+    publisher: 'Logged in user',
+    description: pathway.description ?? `Pathway: ${pathway.name}`,
+    library: [`urn:uuid:Library/${libraryId}`],
+    action: []
+  };
+
+  return strategyDefinition;
+}
+
+function createRecommendationDefinition(node: ActionNode): PlanDefinition {
+  const planId = shortid.generate();
+
+  const recommendationDefinition: PlanDefinition = {
+    id: planId,
+    resourceType: 'PlanDefinition',
+    meta: {
+      profile: ['http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-recommendationdefinition']
+    },
+    extension: [
+      {
+        url: 'http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-knowledgeCapability',
+        valueCode: 'executable'
+      }
+    ],
+    url: `urn:uuid:PlanDefinition/${planId}`,
+    version: '1.0',
+    name: planId,
+    title: node.label,
+    type: {
+      coding: [
+        {
+          system: 'http://terminology.hl7.org/CodeSystem/plan-definition-type',
+          code: 'workflow-definition',
+          display: 'workflow-definition'
+        }
+      ]
+    },
+    status: PLANDEFINITION_DRAFT,
+    experimental: true,
+    publisher: 'Logged in user',
+    description: `Represents an action for ${node.label}`,
+    action: []
+  };
+
+  return recommendationDefinition;
+}
+
+function createBundleEntry(resource: PlanDefinition | ActivityDefinition | Library): BundleEntry {
+  return {
+    fullUrl: `urn:uuid:${resource.resourceType}/${resource.id}`,
+    resource: resource
+  };
+}
+
+function createLibrary(pathway: Pathway): Library {
+  const libraryId = shortid.generate();
+
+  const library: Library = {
+    id: libraryId,
+    resourceType: 'Library',
+    meta: {
+      profile: ['http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-executablelibrary']
+    },
+    extension: [
+      {
+        url: 'http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-knowledgeCapability',
+        valueCode: 'executable'
+      }
+    ],
+    url: `urn:uuid:Library/${libraryId}`,
+    version: '1.0',
+    name: libraryId,
+    title: `Library for ${pathway.name}`,
+    status: LIBRARY_DRAFT,
+    experimental: true,
+    type: {
+      coding: [
+        {
+          system: '',
+          code: '',
+          display: ''
+        }
+      ]
+    },
+    publisher: 'Logged in user',
+    description: `ELM library for pathway: ${pathway.name}`,
+    content: []
+  };
+
+  if (pathway.elm) {
+    library.content.push(
+      {
+        id: 'navigational',
+        contentType: '',
+        data: btoa(JSON.stringify(pathway.elm.navigational)),
+        title: 'ELM for navigating the pathway'
+      },
+      {
+        id: 'precondition',
+        contentType: '',
+        data: btoa(JSON.stringify(pathway.elm.preconditions)),
+        title: 'ELM for pathway preconditions'
+      }
+    );
+  }
+
+  return library;
+}
+
+export function toCPG(pathway: Pathway): Bundle {
+  const bundle: Bundle = {
+    id: pathway.id,
+    resourceType: 'Bundle',
+    type: BUNDLE_COLLECTION,
```

Comment:
Super minor, but could these be made "transaction" type bundles? That makes them a lot easier to POST to a server. The other thing you would need to do is add a `request: { method: 'POST', url: '/{resourceType}' }` to each entry.